apps/prairielearn/src/pages/instructorStudents/components/StudentsCard.tsx

```diff
@@ -0,0 +1,145 @@
+const DEFAULT_SORT: SortingState = [];
+
+export function StudentsCard({ students }: { students: StudentRow[] }) {
+  const [globalFilter, setGlobalFilter] = useQueryState('search', parseAsString.withDefault(''));
+  const [sorting, setSorting] = useQueryState<SortingState>(
+    'sort',
+    parseAsSortingState.withDefault(DEFAULT_SORT),
+  );
+
+  const [columnFilters, setColumnFilters] = useState<ColumnFiltersState>([]);
+  const [columnSizing, setColumnSizing] = useState<ColumnSizingState>({});
+  const [columnVisibility, setColumnVisibility] = useState<ColumnVisibilityState>({});
+  const [columnPinning, setColumnPinning] = useState<ColumnPinningState>({
+    left: [],
+    right: [],
+  });
+
+  const columns = useMemo<ColumnDef<StudentRow, any>[]>(
+    () => [
+      columnHelper.accessor('uid', {
+        header: 'UID',
+        cell: (info) => info.getValue(),
+      }),
+      columnHelper.accessor('name', {
+        header: 'Name',
+        cell: (info) => info.getValue() || '—',
+      }),
+      columnHelper.accessor('email', {
+        header: 'Email',
+        cell: (info) => info.getValue() || '—',
+      }),
+      columnHelper.accessor('created_at', {
+        header: 'Enrolled At',
+        cell: (info) => {
+          const date = new Date(info.getValue());
+          return date.toLocaleDateString() + ' ' + date.toLocaleTimeString();
+        },
+      }),
+    ],
+    [],
+  );
+
+  const table = useReactTable({
+    data: students,
+    columns,
+    columnResizeMode: 'onChange',
+    getRowId: (row) => row.uid,
+    state: {
+      sorting: sorting ?? undefined,
+      columnFilters,
+      globalFilter,
+      columnSizing,
+      columnVisibility,
+      columnPinning,
+    },
+    onSortingChange: setSorting,
+    onColumnFiltersChange: setColumnFilters,
+    onGlobalFilterChange: setGlobalFilter,
+    onColumnSizingChange: setColumnSizing,
+    onColumnVisibilityChange: setColumnVisibility,
+    onColumnPinningChange: setColumnPinning,
+    getCoreRowModel: getCoreRowModel(),
+    getSortedRowModel: getSortedRowModel(),
+    getFilteredRowModel: getFilteredRowModel(),
+    defaultColumn: {
+      minSize: 150,
+      size: 300,
+      maxSize: 500,
+      enableSorting: true,
+      enableHiding: true,
+      enablePinning: true,
+    },
+  });
+
+  return (
+    <div class="card d-flex flex-column h-100">
+      <div class="card-header bg-primary text-white">
+        <div class="d-flex align-items-center justify-content-between">
+          <div>Students</div>
+          <div>
+            <DownloadButton students={students} table={table} />
+          </div>
+        </div>
+      </div>
+      <div class="card-body d-flex flex-column">
+        <div class="d-flex flex-row mb-2">
+          <div class="col-xl-4 col-md-6 col-8 col-auto">
+            <input
+              type="text"
+              id="search-input"
+              class="form-control"
+              placeholder="Search by UID, name, email..."
```

Comment:
This input needs an `aria-label`. I'm a little surprised that our accessibility tests didn't pick up on that...

Reply:
Changed in 6bfdc21e358d030e70cbbb0b61c07a8022d3ed17. Why would the accessibility tests not pick this up?